### ISSUE TITLE
`ForecasterAutoreg` fails to fit when index do not start from 0

### ISSUE THREAD

**Issue by yarnabrina** [JoaquinAmatRodrigo/skforecast#576]
## MCVE

```python
import numpy
import pandas

numpy.random.seed(seed=0)

data = pandas.DataFrame(
    numpy.random.random(size=3 * 20).reshape((20, 3)),
    index=numpy.arange(3, 3 + 20),  # commenting this line will make it work
    columns=["y", "x1", "x2"],
)

from sklearn.linear_model import LinearRegression
regressor = LinearRegression()

from skforecast.ForecasterAutoreg import ForecasterAutoreg
forecaster = ForecasterAutoreg(regressor, 1)

forecaster.fit(data.iloc[:, 0], exog=data.iloc[:, 1:])
```

## Error

> ValueError: Input X contains NaN.
LinearRegression does not accept missing values encoded as NaN natively. For supervised learning, you might want to consider sklearn.ensemble.HistGradientBoostingClassifier and Regressor which accept missing values encoded as NaNs natively. Alternatively, it is possible to preprocess the data, for instance by using an imputer transformer in a pipeline or drop samples with missing values. See https://scikit-learn.org/stable/modules/impute.html You can find a list of all estimators that handle NaN values at the following page: https://scikit-learn.org/stable/modules/impute.html#estimators-that-handle-nan-values

## Expectation

The expectation was that it should work successfully, as passed endogenous and exogenous variables still have same index.

(Found as part of https://github.com/sktime/sktime/pull/5447)

**Comment by JavierEscobarOrtiz** [comment]
Hello @yarnabrina

Thank you for opening the issue!

We have identified the root of the problem. What I can recommend you to avoid this error is to use a `pd.RangeIndex` object instead of a `numpy.arange()`. This will allow you to use an index that doesn't start at 0.

```python
import numpy as np
import pandas as pd

np.random.seed(seed=0)

data = pd.DataFrame(
    np.random.random(size=3 * 20).reshape((20, 3)),
    index=pd.RangeIndex(3, 3 + 20), # Changed to pd.RangeIndex()
    columns=["y", "x1", "x2"],
)

from sklearn.linear_model import LinearRegression
regressor = LinearRegression()

from skforecast.ForecasterAutoreg import ForecasterAutoreg
forecaster = ForecasterAutoreg(regressor, 1)

forecaster.fit(data.iloc[:, 0], exog=data.iloc[:, 1:])
```

We will have a look 😄 

Thank you!

**Comment by JavierEscobarOrtiz** [comment]
Hi @yarnabrina 

This bug is fixed in version 0.11.0. Although this version is not yet released in PyPI, you can test it by installing skforecast from GitHub.

```bash
pip install git+https://github.com/JoaquinAmatRodrigo/skforecast@0.11.x
```

Hope it helps!

**Comment by JavierEscobarOrtiz** [comment]
Skforecast 0.11.0 has been released in PyPI.
